src/blocks/remote-data-html/edit.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	transformStyles,
+	store as blockEditorStore,
+	EditorStyle,
+	BlockEditorStoreSelectors,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { SandBox, Placeholder } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useEffect, useMemo } from '@wordpress/element';
+
+import { __ } from '@/utils/i18n';
+
+import './editor.scss';
+
+/**
+ * This block is based on the official core/html block from Gutenberg.
+ * The main difference is that this block binds to HTML from remote data block context instead of user input.
+ */
+
+// Default styles used to unset some of the styles
+// that might be inherited from the editor style.
+const DEFAULT_STYLES = `
+	html,body,:root {
+		margin: 0 !important;
+		padding: 0 !important;
+		overflow: visible !important;
+		min-height: auto !important;
+	}
+`;
+
+export default function RemoteDataHTML(
+	props: BlockEditProps< RemoteDataInnerBlockAttributes >
+): JSX.Element {
+	const { attributes, setAttributes, isSelected } = props;
+	const blockProps = useBlockProps();
+
+	// Pass the content provided by bindings to save() via setAttributes with changed content
+	const { content } = attributes;
+	useEffect( () => {
+		setAttributes( { content } );
+	}, [ content, setAttributes ] );
+
+	const settingStyles = useSelect< BlockEditorStoreSelectors, EditorStyle[] >(
+		select => select( blockEditorStore ).getSettings().styles,
+		[]
+	);
+
+	const styles = useMemo(
+		() => [
+			DEFAULT_STYLES,
+			...transformStyles( ( settingStyles ?? [] ).filter( ( style: EditorStyle ) => style.css ) ),
+		],
+		[ settingStyles ]
+	);
+
+	const hasBindings = attributes?.metadata?.bindings?.content !== undefined;
+
+	const contentString = content?.toString() ?? '';
+	const hasContent = contentString.length > 0;
+
+	if ( ! hasBindings && ! hasContent ) {
+		return (
+			<div { ...blockProps }>
+				<PlaceholderInstructions />
+			</div>
+		);
+	}
+
+	return (
+		<div { ...blockProps }>
+			<SandBox
+				html={ contentString }
+				styles={ styles }
+				title={ __( 'Remote Data Block HTML Preview' ) }
+				tabIndex={ -1 }
+			/>
+			{
+				/* Similar to core/html, add an overlay register click events. */
```

Comment:
Could you expand on why this is useful? To prevent interaction?

Reply:
It's to allow interaction in the editor, mostly. I've expanded that comment here: https://github.com/Automattic/remote-data-blocks/commit/ff57a4db2069e9feb27a8546f158a45497dcc214